src/journal/journald-context.c

```diff
@@ -500,6 +513,70 @@ static int client_context_read_log_ratelimit_burst(ClientContext *c) {
         return safe_atou(value, &c->log_ratelimit_burst);
 }
 
+static int client_context_read_log_regex(
+                ClientContext *c,
+                const char *key,
+                char **regex,
+                pcre2_code **compiled_regex) {
+#if HAVE_PCRE2
+        int r;
```

Comment:
We always define `r` on the last line of the variable declarations at the start of a function